include/RAJA/policy/cuda/reduce.hpp

```diff
@@ -603,20 +603,26 @@ struct Reduce_Data {
   RAJA::detail::SoAPtr<T, device_mempool_type> device;
   bool own_device_ptr;
 
-  //! disallow default constructor
-  Reduce_Data() = delete;
+  Reduce_Data()
+  {
+    reset(T(), T());
+  }
 
   /*! \brief create from a default value and offload information
    *
    *  allocates PinnedTally to hold device values
    */
-  explicit Reduce_Data(T initValue, T identity_)
-      : value{initValue},
-        identity{identity_},
-        device_count{nullptr},
-        device{},
-        own_device_ptr{false}
+  Reduce_Data(T initValue, T identity_)
   {
+    reset(initValue, identity_);
+  }
+
+  void reset(T initValue, T identity_ = T()){
```

Comment:
I wasn't initially sure this would work -- so long as the CUDA tests pass, LGTM 👍